File: SlidingWindow/MaxProfit.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int start = 0;
        int end = 1;
        while(end < prices.size())
        {
            if(prices[end] - prices[start] > 0)
                profit = max(profit,prices[end] - prices[start]);
            else
                start = end;
            end++;
        }
        return profit;
    }
};

File: SlidingWindow/MaxProfit.py
lass Solution(object):
    def maxProfit(self, prices):
        profit = 0
        start, end = 0, 1
        while end < len(prices):
            if prices[end] - prices[start] > 0:
                profit = max(profit , prices[end] - prices[start])
            else:
                start = end
            end += 1
        return profit